### ISSUE TITLE
Maxperf profile is broken on Windows

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#3964]
## Description

The `maxperf` profile results in a linking error on Windows due to a very strange bug either in `rustc`, LLVM or the unsafe code of a dependency (which causes the miscompilation). The bug is being tracked here: https://github.com/rust-lang/rust/issues/107781

We have no way to fix it, so the workaround is to compile with the release profile on Windows, which is the default for users and will be the default for our release binaries as of v3.5.0.

## Version

Lighthouse v3.4.0

**Comment by xanatos** [comment]
Question: from the description of the 3.50 release it seems that it is still compatible with Rust 1.65, and from the description of the rust-lang bug, it seems that Rust 1.65 and 1.66.1 aren't affected by the bug. Wouldn't it be possible to generate the maxperf binaries for Windows with one of those versions of Rust?

**Comment by michaelsproul** [comment]
@xanatos Yeah it would, I guess we could do that for the next release. It just requires a little bit of fiddling with the release CI.

**Comment by michaelsproul** [comment]
Can confirm that this is fixed in Rust 1.70.0 nightly. Once that comes out around June 1st we can re-enable `maxperf`.

**Comment by michaelsproul** [comment]
Can confirm that this is fixed with the Rust 1.70.0 release. I compiled successfully with `maxperf` just now:

```
Lighthouse v4.2.0-7b1a0d4
BLS library: blst-modern
SHA256 hardware acceleration: true
Allocator: system
Profile: maxperf
Specs: mainnet (true), minimal (false), gnosis (false)
```

**Comment by michaelsproul** [comment]
Resolved by #4371 and tested with the v4.2.990 tree-states release
